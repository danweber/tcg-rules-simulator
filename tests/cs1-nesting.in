VERSION:2
# For stepping through nested effects.

# P2 attacks, triggering:
## trigger 1: their own [WA] and opponents "when being attacked"
# The first [WA] happens, deleting something, triggering:
## trigger 2: that thing's [on deletion] and P2's "when a monster is deleted"
# The [on deletion] happens, playing something out. This triggers:
## trigger 3: two things on P1's side (neither of which will have an effect, but both still trigger)


FIELD1:WhenBeingAtt,CS1-05 WhenPlay1,CS1-06 WhenPlay2,CS1-06 Phoenix1,CS1-Phoenix Phoenix2,CS1-Phoenix
FIELD2:WhenAtt,DW1-Van,CS2-05 
SECURITY1:DW1-Van
GAME:P2 T3 MAIN 6
DUMP:DP

CMD:ATTACK WhenAtt Player

STEP:TO:0
DUMP:DP

ECHO: --- should be three levels deep
EFFECT-TREE

# activate useless effects.
# First, pick one:
STEP:TO:0
# Now, say yes:
STEP:TO:0
# Now, say yes to the other:

STEP:TO:0

ECHO: -- 
DUMP:DP

ECHO: -- one 1 effect left
# we should be waiting for the Ice Shield effect from step 1 now
EFFECT-TREE

ECHO: -- finish it off
STEP:AUTOANSWER:CAPTURE
DUMP
EXIT


