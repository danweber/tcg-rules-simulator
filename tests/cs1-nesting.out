FIELD1:WhenBeingAtt,CS1-05,6500DP WhenPlay1,CS1-06,7000DP WhenPlay2,CS1-06,7000DP Phoenix1,CS1-08,1000DP Phoenix2,CS1-08,1000DP
FIELD2:WhenAtt,DW1-Vanillamon,CS2-05,5000DP
GAME:P2 T3 MAIN 6
FIELD1:WhenBeingAtt,CS1-05,6500DP WhenPlay1,CS1-06,7000DP WhenPlay2,CS1-06,7000DP Phoenix2,CS1-08,1000DP
FIELD2:WhenAtt,DW1-Vanillamon,CS2-05,REST,5000DP
GAME:P2 T3 MAIN 6
ECHO: --- should be three levels deep
EFFECT-TREE:DSE DEPTH 1 DONE:suspend to attack, DONE:attack declare
EFFECT-TREE:RES DEPTH 2 Ice Shield, DONE:Great Sword
EFFECT-TREE:RES DEPTH 3 PROCESSING:Phoenix Defender
EFFECT-TREE:INT DEPTH 3 Sneaky Cat, Sneaky Cat
ECHO: --
FIELD1:WhenBeingAtt,CS1-05,6500DP WhenPlay1,CS1-06,REST,7000DP WhenPlay2,CS1-06,REST,7000DP Phoenix2,CS1-08,1000DP I7,CS1-08,1000DP
FIELD2:WhenAtt,DW1-Vanillamon,CS2-05,REST,5000DP
GAME:P2 T3 MAIN 6
ECHO: -- one 1 effect left
EFFECT-TREE:DSE DEPTH 1 DONE:suspend to attack, DONE:attack declare
EFFECT-TREE:RES DEPTH 2 PROCESSING:Ice Shield
ECHO: -- finish it off
CAPTURE:Activate Ice Shield suspending
CAPTURE:Don't pay/activate
CAPTURE:Target Ice Shield 1
CAPTURE:Target Sneaky Cat 2
CAPTURE:Target Sneaky Cat 3
CAPTURE:Target Phoenix Defender 5
CAPTURE:Target VanillaMon 6
CAPTURE:Target Phoenix Defender 7
FIELD1:WhenBeingAtt,CS1-05,REST WhenPlay1,CS1-06,REST WhenPlay2,CS1-06,REST Phoenix2,CS1-08 I7,CS1-08
FIELD2:
GAME:P2 T3 MAIN 6
